### DISCUSSION TITLE
useKeyPress是否支持target延迟绑定或动态绑定

### DISCUSSION THREAD

**Discussion by ida-fe** [alibaba/hooks#1474] [Q&A]
调用useKeyPress时可能target的dom还没创建完毕，是否支持hooks内监听target存在时，再进行事件绑定

**Comment by brickspert** [comment]
v3 版本的 target 支持 target 动态变化的，也就是会处理后创建的 dom.

不过在没有 dom 时候，是默认 window 的.

你是期望没有 target 的时候，也不要默认 window？

**Reply by ida-fe** [reply]
嗯对的，希望没有target时，可以判断不要触发handler

**Reply by brickspert** [reply]
建议当前先在回调函数里面，手动判断下目标 target 有没有，以做处理。

至于你说的 feature，可能需要加一个 options，来控制是否需要默认 window 了。
